modules/integration/src/test/scala/scala/cli/integration/NativePackagerTests.scala

```diff
@@ -202,4 +202,80 @@ class NativePackagerTests extends munit.FunSuite {
         expect(os.isFile(launcher))
       }
     }
+
+  test("building docker image") {
+
+    testInputs.fromRoot { root =>
+
+      val appName         = helloWorldFileName.stripSuffix(".scala")
+      val imageRepository = appName.toLowerCase()
+      val imageTag        = "latest"
+
+      // format: off
+      val cmd = Seq[os.Shellable](
+        TestUtil.cli,
+        "package", helloWorldFileName,
+        "--docker",
+        "--docker-from", "adoptopenjdk/openjdk8",
+        "--docker-image-repository", imageRepository,
+        "--docker-image-tag", imageTag
+      )
+      // format: on
+
+      os.proc(cmd)
+        .call(
+          cwd = root,
+          stdin = os.Inherit,
+          stdout = os.Inherit
+        )
+
+      val expectedImage =
+        s"$imageRepository:$imageTag"
+
+      val output = os.proc("docker", "run", expectedImage).call(cwd = os.root).out.text.trim
+      expect(output == message)
+
+      // clear
+      os.proc("docker", "rmi", "-f", expectedImage).call(cwd = os.root)
+    }
+  }
+
+  test("building docker image with scala.js app") {
+
+    testInputs.fromRoot { root =>
+
+      val appName         = helloWorldFileName.stripSuffix(".scala")
+      val imageRepository = appName.toLowerCase()
+      val imageTag        = "latest"
+
+      // format: off
+      val cmd = Seq[os.Shellable](
+        TestUtil.cli,
+        "package", helloWorldFileName,
+        "--js",
+        "--docker",
+        "--docker-from", "node",
+        "--docker-image-repository", imageRepository,
+        "--docker-image-tag", imageTag
+      )
+      // format: on
+
+      os.proc(cmd)
+        .call(
+          cwd = root,
+          stdin = os.Inherit,
+          stdout = os.Inherit
+        )
+
+      val expectedImage =
+        s"$imageRepository:$imageTag"
+
+      val output = os.proc("docker", "run", expectedImage).call(cwd = os.root).out.text.trim
+      expect(output == message)
+
+      // clear
+      os.proc("docker", "rmi", "-f", expectedImage).call(cwd = os.root)
```

Comment:
Same as above: this could be done from a `finally` clause.

Reply:
Thanks for the review. Right, it will be a better solution.